torch_np/_detail/implementations.py

```diff
@@ -118,3 +371,101 @@ def meshgrid(*xi_tensors, copy=True, sparse=False, indexing="xy"):
         output = [x.clone() for x in output]
 
     return output
+
+
+def bincount(x_tensor, /, weights_tensor=None, minlength=0):
+    int_dtype = _dtypes_impl.default_int_dtype
+    (x_tensor,) = _util.cast_dont_broadcast((x_tensor,), int_dtype, casting="safe")
+
+    result = torch.bincount(x_tensor, weights_tensor, minlength)
+    return result
+
+
+# ### linspace, geomspace, logspace and arange ###
+
+
+def geomspace(start, stop, num=50, endpoint=True, dtype=None, axis=0):
+    if axis != 0 or not endpoint:
+        raise NotImplementedError
+    tstart, tstop = torch.as_tensor([start, stop])
```

Comment:
Why do we have this line? Shouldn't these be tensors already here?

Reply:
```
TypeError: pow() received an invalid combination of arguments - got (float, float), but expected one of:
 * (Tensor input, Tensor exponent, *, Tensor out)
 * (Number self, Tensor exponent, *, Tensor out)
 * (Tensor input, Number exponent, *, Tensor out)
```

is a one of those counter-examples of pytorch being mostly consistent :-).

Will add conversion to tensors to the wrapper to make the internt clearer then.

Reply:
PyTorch functions never accept two numbers. If this gets two numbers, why don't you use `a ** b`? You'll need to do the same thing by choosing between `torch.log` and `math.log`.